File: guides/micronaut-logstash/java/src/main/java/example/micronaut/Bootstrap.java
package example.micronaut;

import io.micronaut.context.event.ApplicationEventListener;
import io.micronaut.context.event.StartupEvent;
import javax.inject.Singleton;
import org.slf4j.LoggerFactory;
import org.slf4j.Logger;

@Singleton // <1>
class Bootstrap implements ApplicationEventListener<StartupEvent> {  // <2>
    private static final Logger LOG = LoggerFactory.getLogger(Bootstrap.class);

    @Override
    public void onApplicationEvent(StartupEvent event) {
        LOG.info("application started");
    }
}

File: guides/micronaut-scheduled/kotlin/src/main/kotlin/example/micronaut/BootStrap.kt
package example.micronaut

import io.micronaut.context.event.ApplicationEventListener
import io.micronaut.runtime.server.event.ServerStartupEvent
import javax.inject.Singleton

@Singleton // <1>
class BootStrap(val registerUseCase: RegisterUseCase) // <2>
    : ApplicationEventListener<ServerStartupEvent> { // <3>

    override fun onApplicationEvent(event: ServerStartupEvent) {  // <4>
        try {
            registerUseCase.register("harry@micronaut.example")
            Thread.sleep(20000)
            registerUseCase.register("ron@micronaut.example")
        } catch (e: InterruptedException) {
            e.printStackTrace()
        }
    }
}
